carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -233,6 +233,7 @@ void WMBroadcaster::addPassingControlLineFromMsg(std::shared_ptr<Geofence> gf_pt
 void WMBroadcaster::addRegionAccessRule(std::shared_ptr<Geofence> gf_ptr, const cav_msgs::TrafficControlMessageV01& msg_v01, const std::vector<lanelet::Lanelet>& affected_llts) const
 {
    gf_ptr->regulatory_element_ = std::make_shared<lanelet::RegionAccessRule>(lanelet::RegionAccessRule::buildData(lanelet::utils::getId(),affected_llts,{},participantsChecker(msg_v01)));
+   route_flag_=true;
```

Comment:
I think it would be safer for this flag to be part of the Geofence object gf_ptr refers to. The addGeofence callback is triggered based on a geofence's  schedule, so its possible you could encounter a race condition where this is set for another geofence that is not yet triggered, but get applied for a different one that is. 

If it is part of the Geofence object this would not be an issue

Reply:
Done